src/App/Fossa/Analyze/Project.hs

```diff
@@ -42,4 +42,5 @@ data ProjectResult = ProjectResult
 
 shouldKeepUnreachableDeps :: Text -> Bool
 shouldKeepUnreachableDeps "swift" = True
+shouldKeepUnreachableDeps "gomod" = True
```

Comment:
I'd love to kill this logic by moving dependency pruning into the tactics that need it (namely, the golang tactics that call `fillInTransitive`). matching on strings is brittle and `fillInTransitive` is the only reason we prune unreachable dependencies anyway

Reply:
Addressing in https://github.com/fossas/fossa-cli/pull/721